src/plugins/intel_cpu/src/scratch_pad.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "common/memory.hpp"
+#include <memory>
+
+namespace ov {
+namespace intel_cpu {
+
+class ScratchPad {
+    size_t m_curScratchpadSize = 0;
+    std::shared_ptr<void> m_curScratchpadMem = nullptr;
```

Comment:
Is not it better to use the plugin's memory implementation? It already can be resized according to a new mem descriptor, also it allocated aligned memory, just the same in this class.

Reply:
The CPU Memory class is not suitable to represent a pure memory concept as required here, for example, `Memory` implies `precision` `layout` `shapes` which would be meaningless and confusing for representing a pure memory region such as scratch pad, so why should we make simple concept more complex by reusing `Memory`?

Reply:
On the one hand you are right, it is better to keep thing simple. But on the other hand, we already pass `dnnl::memory::desc & md` to the `setScratchPad` method See L21, so we can simply use this memory descriptor to resize the Memory object (`redefineMemoryDesc` call), everything will be handled inside.

Reply:
OK, Fixed